zegami_sdk/annotation.py

```diff
@@ -340,3 +340,53 @@ def base64_to_boolmask(b64_data):
                            np.array(img), dtype='uint8')
 
         return arr_int > 127
+
+
+class AnnotationBB(_Annotation):
+    """
+    An annotation comprising a bounding box and some metadata.
+
+    # TODO
+    To view the bounding boxes of an image, use bb.view().
+
+    Note: Providing imageset_id and image_index is not mandatory and can be
+    obtained automatically, but this is slow and can cause unnecessary
+    re-downloading of data.
+    """
+
+    TYPE = 'zc-boundingbox'
+    UPLOADABLE_DESCRIPTION = """
+        Bounding box annotation data includes the bounding box bounds,
+        a width and height, and score if generated
+        by a model (else None).
+    """
+
+    @classmethod
+    def create_uploadable(cls, bounds, class_id):
+        """
+        Creates a data package ready to be uploaded with a collection's
+        .upload_annotation().
+
+        Input 'bounds' is a dictionary of { height, width, x, y } for the
+        edges of the given bounding box.
+
+        Note: The output of this is NOT an annotation, it is used to upload
+        annotation data to Zegami, which when retrieved will form an
+        annotation.
+        """
+
+        data = {
+            'type': cls.TYPE,
+            'x': int(bounds['x']),
```

Comment:
Maybe it should be the user's responsibility to convert them ints (if that matters for them), rather than rounding it here.
The ones created in the client are floats and can start partway across a pixel.